tests/basis_test.py

```diff
@@ -447,10 +763,466 @@ def test_cheb_2nd_call_root_points():
     assert numpy.allclose(f(root_points), numpy.zeros(3))
 
 
+def test_is_complex():
+    """Test that basis functions that output complex values indicate it"""
+    assert smolyay.basis.ChebyshevFirstKind(0)._is_complex == False
+    assert smolyay.basis.ChebyshevSecondKind(0)._is_complex == False
+    assert smolyay.basis.Trigonometric(0)._is_complex == True
+
+
 # Test a set of basis functions
-def test_set_initialize():
-    """Check BasisFunctionSet correctly initializes"""
-    f = smolyay.basis.BasisFunctionSet([])
-    f2 = smolyay.basis.BasisFunctionSet([smolyay.basis.ChebyshevFirstKind(0)])
-    assert f.basis_functions == []
-    assert f2.basis_functions[0].degree == 0
+def test_set_base_class_initialize():
+    """Test BasisFunctionSet correctly initializes"""
+    f = smolyay.basis.BasisFunctionSet([smolyay.basis.ChebyshevFirstKind(0)])
+    assert f.basis_functions[0].degree == 0
+    assert f[0].degree == 0
+    assert len(f) == 1
+    assert numpy.array_equal(f.domain, [-1, 1])
+    f = smolyay.basis.BasisFunctionSet([smolyay.basis.Trigonometric(0)])
+    assert f.basis_functions[0].frequency == 0
+    assert f[0].frequency == 0
+    assert len(f) == 1
+    assert numpy.array_equal(f.domain, [0, 2 * numpy.pi])
+
+
+def test_set_base_class_initialize_error():
+    """Test BasisFunctionSet gives error if basis function have different domains"""
+    with pytest.raises(AttributeError):
+        f = smolyay.basis.BasisFunctionSet([])
+        f.domain
+    with pytest.raises(AttributeError):
+        f = smolyay.basis.BasisFunctionSet([])
+        f.scale_to_domain([1, 2, 3, 4], [0, 5])
+    with pytest.raises(TypeError):
+        smolyay.basis.BasisFunctionSet(
+            [smolyay.basis.ChebyshevFirstKind(0), smolyay.basis.Trigonometric(0)]
+        )
+    with pytest.raises(TypeError):
+        smolyay.basis.BasisFunctionSet(
+            [smolyay.basis.ChebyshevSecondKind(0), smolyay.basis.Trigonometric(0)]
+        )
+    with pytest.raises(TypeError):
+        smolyay.basis.BasisFunctionSet(
+            [smolyay.basis.ChebyshevFirstKind(0), smolyay.basis.ChebyshevSecondKind(0)]
+        )
+
+
+@pytest.mark.parametrize(
+    "basis_set",
+    [
+        smolyay.basis.ChebyshevFirstKindBasisFunctionSet,
+        smolyay.basis.ChebyshevSecondKindBasisFunctionSet,
+    ],
+    ids=[
+        "ChebyshevFirstKind",
+        "ChebyshevSecondKind",
+    ],
+)
+def test_set_cheb_initialize(basis_set):
+    """Test Chebyshev function sets correctly initialize"""
+    f = basis_set(3)
+    assert f.basis_functions[0].degree == 0
+    assert f[0].degree == 0
```

Comment:
You don't need to check both of these accessors if you have confirmed the accessor is working elsewhere.